### PR TITLE
feat(components): create new Tag component

### PR DESCRIPTION
**Pull request by jerader**
closes RSQ-84 https://opentrons.atlassian.net/browse/RSQ-84

# Overview

This PR creates the Tag component that was added to the design system. Additionally, it deprecates the usage of the `basic` type in `Chip` component

# Test Plan

Search for `Tag` in Storybook and look through the different props. Make sure it matches the Figma designs

# Changelog

- create new Tag component folder and create the component and storybook
- remove `basic` type from Chip component, story and test
NOTE: i did not make a test for this component. I think atoms in the components directory like this are not as important to have a test since we can check how it looks/works via the story

# Review requests

see test plan

# Risk assessment

low

### CODE REVIEW

File: components/src/atoms/Tag/index.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+          : DEFAULT_CONTAINER_STYLE
+      }
+      gridGap={SPACING.spacing4}
+      data-testid={`Tag_${type}`}
+      {...styleProps}
+    >
+      {iconName != null && iconPosition === 'left' ? (
+        <Icon name={iconName} aria-label={`icon_${text}`} css={ICON_STYLE} />
+      ) : null}
+      <StyledText css={TEXT_STYLE}>{text}</StyledText>
```
Comment (ncdiehl11) [inline components/src/atoms/Tag/index.tsx:105]: nit: can we use `as='h3'` to avoid redundancy?

**Review by ncdiehl11 — APPROVED** [review]
looks good with one nit!

File: components/src/atoms/Tag/index.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    backgroundColor: COLORS.blue50,
+    color: COLORS.white,
+  },
+}
+
+export function Tag(props: TagProps): JSX.Element {
+  const { iconName, type, text, iconPosition, ...styleProps } = props
+
+  const DEFAULT_CONTAINER_STYLE = css`
+    padding: 2px 8px;
```
Comment (koji) [inline components/src/atoms/Tag/index.tsx]:
```suggestion
    padding: ${SPACING.spacing2} ${SPACING.spacing8};
```

File: components/src/atoms/Tag/index.tsx (hunk 1 above)
Comment (jerader) [reply components/src/atoms/Tag/index.tsx:105]: i think that the `h3` `StyledText` tag doesn't translate correctly for the ODD so i'd have to override it still with the css prop

File: components/src/atoms/Tag/index.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    border-radius: ${BORDERS.borderRadius4};
+    @media ${RESPONSIVENESS.touchscreenMediaQuerySpecs} {
+      border-radius: ${BORDERS.borderRadius8};
+      padding: 8px 12px;
```
Comment (koji) [inline components/src/atoms/Tag/index.tsx]:
```suggestion
      padding: ${SPACING.spacing8} ${SPACING.spacing12};
```

File: components/src/atoms/Tag/index.tsx (hunk 2 above)
Comment (jerader) [reply components/src/atoms/Tag/index.tsx]: oops! i forgot to change those to the const, thanks!

File: components/src/atoms/Tag/index.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        type === 'interactive'
+          ? INTERACTIVE_CONTAINER_STYLE
+          : DEFAULT_CONTAINER_STYLE
+      }
+      gridGap={SPACING.spacing4}
+      data-testid={`Tag_${type}`}
+      {...styleProps}
+    >
+      {iconName != null && iconPosition === 'left' ? (
+        <Icon name={iconName} aria-label={`icon_${text}`} css={ICON_STYLE} />
```
Comment (koji) [inline components/src/atoms/Tag/index.tsx]: nit

for test adding `left` might be useful?

```suggestion
        <Icon name={iconName} aria-label={`icon_left_${text}`} css={ICON_STYLE} />
```

**Comment by koji** [comment]
@jerader 

Could you add tests for Tag component?

I think need to add Tag to `components/src/atoms/index.ts`

File: components/src/atoms/Tag/Tag.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
+import * as React from 'react'
+
+import { Flex } from '../../primitives'
+import { SPACING, VIEWPORT } from '../../ui-style-constants'
+import { Tag as TagComponent } from './index'
+import type { Meta, StoryObj } from '@storybook/react'
+
+const meta: Meta<typeof TagComponent> = {
+  title: 'Library/Atoms/Chip',
```
Comment (koji) [inline components/src/atoms/Tag/Tag.stories.tsx]:
```suggestion
  title: 'Library/Atoms/Tag',
```

File: components/src/atoms/Tag/index.tsx [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+  const INTERACTIVE_CONTAINER_STYLE = css`
+    ${DEFAULT_CONTAINER_STYLE}
+    &:hover {
+      background-color: ${COLORS.black90}${COLORS.opacity40HexCode};
+    }
+    &:focus-visible {
+      background-color: ${COLORS.black90}${COLORS.opacity40HexCode};
+      box-shadow: 0 0 0 3px ${COLORS.blue50};
+    }
+  `
```
Comment (koji) [inline components/src/atoms/Tag/index.tsx]:
```suggestion
    &:focus-visible {
      box-shadow: 0 0 0 3px ${COLORS.blue50};
      outline: none;
    }
  `
```

when i turned on `focus-visible` status on dev tools.
2024-06-18 00_29_37-Library _ Atoms _ Chip - Tag ⋅ Storybook and 10 more pages - Personal - Microsof



2024-06-18 00_28_52-Helix Design System – Figma

File: components/src/atoms/Tag/__tests__/Tag.test.tsx
Comment (koji) [inline components/src/atoms/Tag/__tests__/Tag.test.tsx:1]: 👍

**Comment by koji** [comment]
@jerader 
recommend you share the storybook with Mel to get her review.

**Comment by jerader** [comment]
> @jerader recommend you share the storybook with Mel to get her review.

@koji yup! i tagged her and left a comment in the ticket with the storybook link. she was ooo yesterday so i'll ping her today to remind her to take a look

**Review by koji — APPROVED** [review]
once you update `focus-within` style and Storybook title, this pr is good to go on dev side.

**Pull request merged**